test/rekt/features/apiserversource/data_plane.go

```diff
@@ -822,6 +823,73 @@ func SendsEventsWithRetries() *feature.Feature {
 	return f
 }
 
+func setupNodeLabels(nodeLabels map[string]string) feature.StepFn {
+	return func(ctx context.Context, t feature.T) {
+		nodes, err := client.Get(ctx).CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+		if err != nil {
+			t.Fatalf("Could not list nodes: %v", err)
+		}
+
+		if len(nodes.Items) == 0 {
+			t.Fatal("No nodes found")
+		}
+
+		randomNode := &nodes.Items[0]
+
+		randomNode.Labels = nodeLabels
+
+		// Update the node with the new labels
+		_, err = client.Get(ctx).CoreV1().Nodes().Update(ctx, randomNode, metav1.UpdateOptions{})
+
+		if err != nil {
+			t.Fatalf("Could not update node: %v", err)
+		}
+	}
+}
+
+func DeployAPIServerSauceWithNodeSelector() *feature.Feature {
+	f := feature.NewFeatureNamed("deploy")
+
+	// same labels as in the configmap
+
+	f.Requirement("setup config-features and load into context", func(ctx context.Context, t feature.T) {
```

Comment:
```suggestion
	f.Setup("setup config-features and load into context", func(ctx context.Context, t feature.T) {
```
If I understand your test correctly you want this done *before* you install the APIServerSource